source/val/BasicBlock.h

```diff
@@ -104,8 +136,12 @@ class BasicBlock {
     /// @brief Constructs an iterator for the given block which points to
     ///        @p block
     ///
-    /// @param block The block which is referenced by the iterator
-    explicit DominatorIterator(const BasicBlock* block);
+    /// @param block          The block which is referenced by the iterator
+    /// @param dominator_func This function will be called to get the dominator
```

Comment:
Do you mean it gets the *immediate* dominator?  Otherwise the choice is not necessarily unique.